editor/src/components/navigator/navigator-item/navigator-item.tsx

```diff
@@ -284,24 +284,32 @@ const isHiddenConditionalBranchSelector = createCachedSelector(
     elementPath: ElementPath,
     parentPath: ElementPath,
   ): boolean => {
+    if (parent == null) {
+      return false
+    }
+    const originalConditionValue = parent.conditionalValue
+    if (originalConditionValue === 'not-a-conditional') {
+      return false
+    }
+
     const conditional = asConditional(parent)
     if (conditional == null) {
       return false
     }
+
     const flag = getConditionalFlag(conditional)
-    return flag === false
-      ? matchesOverriddenBranch(elementPath, parentPath, {
-          clause: conditional.whenTrue,
-          branch: 'then',
-          wantOverride: true,
-          parentOverride: true,
-        })
-      : matchesOverriddenBranch(elementPath, parentPath, {
-          clause: conditional.whenFalse,
-          branch: 'else',
-          wantOverride: false,
-          parentOverride: false,
-        })
+
+    // the final condition value, either from the original or from the override
+    const overriddenConditionValue = flag ?? originalConditionValue
```

Comment:
Might be worth constraining this to `boolean` just in case anything here changes type.